### ISSUE TITLE
Review <Map> element in Jetty XML

### ISSUE THREAD

**Issue by sbordet** [eclipse/jetty.project#5316]
**Jetty version**
10.0.x

**Description**
The `<Map>` element is not used in Jetty XML files.

It cannot be customized to create `Map` implementations, it only creates `HashMap`s.

`<Map>` takes a list of `<Entry>` elements, but `<Entry>` takes a list of 2 `<Item>`, which is error prone (e.g. pass a number of items different from 2).

There is `<Put>` already that however is constrained to have a literal key in the `name` attribute (e.g. the key cannot be the result of a `<SystemProperty>`), which works with any `Map` implementations, for example `java.util.Properties`.

In summary, I think `<Map>` should be removed. Specific implementations can be created with `<New>`, and `<Put>` covers most cases. For the non covered cases, we can always use `<Call>`.

**Comment by gregw** [comment]
I don't think we can remove it without deprecating it for some time first.
Also I'd like to see `Put` enhanced  to support:
```xml
  <Put>
    <Key>name</Key>
    <Value>value</Value>
  </Put>
```
Which is on trend for removing values from element parameters anyway!

If we can do that quickly.. then maybe we can just remove Map and have a configuration-10.dtd

BUT IT HAS TO BE DONE TODAY if it is to make it into 10.0.x

**Comment by sbordet** [comment]
Added `class` attribute and `Class` element to `Map` so that the implementation can be specified.

The other proposed changes were either backwards incompatible or too complex.
